src/Content/Version.php

```diff
@@ -125,61 +163,74 @@ public function modified(
 	public function move(string $fromLanguage, VersionId $toVersionId, string $toLanguage): void
 	{
 		$this->ensure($fromLanguage);
-		$this->model->storage()->move($this->id, $fromLanguage, $toVersionId, $toLanguage);
+		$this->model->storage()->move(
+			fromVersionId: $this->id,
+			fromLanguage: $this->language($fromLanguage),
+			toVersionId: $toVersionId,
+			toLanguage: $this->language($toLanguage)
+		);
 	}
 
 	/**
 	 * Returns the stored content fields
 	 *
-	 * @param string $lang Code `'default'` in a single-lang installation
 	 * @return array<string, string>
 	 */
 	public function read(string $language = 'default'): array
 	{
 		$this->ensure($language);
-		return $this->model->storage()->read($this->id, $language);
+		return $this->model->storage()->read($this->id, $this->language($language));
 	}
 
 	/**
 	 * Updates the modification timestamp of an existing version
 	 *
-	 * @param string $lang Code `'default'` in a single-lang installation
+	 * @param string|null $language If null, all available languages will be touched
 	 *
 	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
 	public function touch(string|null $language = null): void
 	{
-		// touch all languages
-		if ($language === null) {
-			foreach ($this->model->kirby()->languages() as $language) {
-				$this->touch($language->code());
-			}
+		// touch a single language
+		if ($this->model->kirby()->multilang() === false) {
+			$this->touchLanguage('default');
+			return;
 		}
 
-		// make sure the version exists
-		$this->ensure($language);
-
-		// touch the default language in single-language mode
-		if ($this->model->kirby()->multilang() === false) {
-			$this->model->storage()->touch($this->id, 'default');
+		// touch a specific language
+		if ($language !== null) {
+			$this->touchLanguage($language);
 			return;
 		}
 
-		// touch a single language
-		$this->model->storage()->touch($this->id, $language);
+		// touch all languages
+		foreach ($this->model->kirby()->languages() as $language) {
+			$this->touchLanguage($language);
```

Comment:
`Note`: This call especially benefits from allowing to pass `Language` objects to `Version` methods instead of the language strings (see above). Otherwise the object would have to be converted to a string and back to an object for each call here.